mesheryctl/pkg/utils/scripts.go

```diff
@@ -6,118 +6,250 @@ import (
 	"os/exec"
 )
 
-// TODO@Aisuko this function is so hard to read and maintain, please refactor it
-func GenerateConfigGKE(configPath, SAName, namespc string) error {
-	script := fmt.Sprintf(`
-	set -e
-	set -o pipefail
-
-	KUBECFG_FILE_NAME="%s"
-	SERVICE_ACCOUNT_NAME=%s
-	NAMESPACE="%s"
-	TARGET_FOLDER=$(dirname ${KUBECFG_FILE_NAME})
-
-	command_exists() {
-		command -v $1 >/dev/null 2>&1
+// GKEConfig holds configuration for GKE script generation
+
+type GKEConfig struct {
+	ConfigPath string
+	SAName     string
+	Namespace  string
+}
+
+// GenerateConfigGKE generates kubernetes config for GKE
+func GenerateConfigGKE(configPath, SAName, namespace string) error {
+	cfg := &GKEConfig{
+		ConfigPath: configPath,
+		SAName:     SAName,
+		Namespace:  namespace,
 	}
-	check_prerequisites() {
-		if ! command_exists jq ; then
-			echo -e "\\nMissing required utility: 'jq'. Please install 'jq' and try again."
-			exit;
-		fi
-		if ! command_exists base64 ; then
-			echo -e "\\nMissing required utility: 'base64'. Please install 'base64' and try again."
-			exit;
-		fi
-		if ! command_exists awk ; then
-			echo -e "\\nMissing required utility: 'awk'. Please install 'awk' and try again."
-			exit;
-		fi
-		if ! command_exists tail ; then
-			echo -e "\\nMissing required utility: 'tail'. Please install 'tail' and try again."
-			exit;
-		fi
+
+	if err := cfg.validate(); err != nil {
+		return ErrInvalidArgument(err)
 	}
-	create_service_account() {
-		echo -e "\\nCreating a service account in ${NAMESPACE} namespace: ${SERVICE_ACCOUNT_NAME}"
-		kubectl create sa "${SERVICE_ACCOUNT_NAME}" --namespace "${NAMESPACE}"
-		kubectl create clusterrolebinding "${SERVICE_ACCOUNT_NAME}" --clusterrole=cluster-admin --serviceaccount=default:"${SERVICE_ACCOUNT_NAME}" --namespace "${NAMESPACE}"
+
+	return cfg.executeScript()
+}
+
+func (c *GKEConfig) validate() error {
+	if c.ConfigPath == "" || c.SAName == "" || c.Namespace == "" {
+		return fmt.Errorf("configPath, SAName, and namespc are required")
```

Comment:
Should create a mesherytl error and return it

you can check https://github.com/lekaf974/meshery/blob/master/mesheryctl/internal/cli/root/design/error.go as example and read the error guide https://docs.meshery.io/project/contributing/contributing-error


ps: Can you proceed kike this for all errors returned